### DISCUSSION TITLE
Error: Could not load file or assembly - (for nested assembly)

### DISCUSSION THREAD

**Discussion by avivs100** [pythonnet/pythonnet#2293] [Q&A]
The repo with all the code https://github.com/avivs100/pythonnetTest
### Environment

-   Pythonnet version: 3.0.3.0
-   Python version: 3.11.3
-   Operating System: win 10
-   .NET Runtime: 7

### Details

-   I created a simple .NET solution with few projects, 2 class lib and one console application (all in .net7).
the davidProj is a class lib project that contains only 1 class with simple properties

the consoleApp1 create a David instance and save it to json file with System.Text.Json nuget 
the Test.Infrastructure.DavidUtils is also a class lib project that have 1 static class and 1 static function 

```c#
public static David DeserializeDavidFromJsonString(string davidJson)
    {
//using the System.Text.Json nuget
        var david = JsonSerializer.Deserialize<David>(davidJson) ?? throw new Exception("cold not map the json string to david object");
        return david;
    }
``` 

    
    I run the .NET app created a JSON file of the David instance and copy the .dll file to new folder and also save the JSON file to that folder.
    
    now I created in this folder a python file and installed the pythonnet with pip.
    and this is the code
    import JSON

    
```python
import clr
clr.AddReference("Test.Infrastructure.DavidUtils")
clr.AddReference("davidProj")

from Test.Infrastructure.DavidUtils import Functions

fullpath = "C:\\Users\\avivshi\\source\\repos\\pythonnetTest\\PythonNetTest\\davidObj.json"
try:
   
    with open(fullpath, "r") as file:
        json_data = json.load(file)
    json_string = json.dumps(json_data)
    print(json_string)
    david = Functions.DeserializeDavidFromJsonString(json_string)
    # Access the properties
    print(f"Name: {david.Name}")
    print(f"Description: {david.Description}")
    print(f"Numbers: {list(david.Numbers)}")
    print(f"Gender: {david.GetGender()}")
    print(f"Type of Gender: {type(david.Gender)}")

except Exception as e:
    print(f"Error: {e}")
```    

    the error is different with any nuget that i work in the dot net,
    if i use the System.Text.Json so this is the error: 
Error: Could not load file or assembly 'System.Memory, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file specified.
File name: 'System.Memory, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at Test.Infrastructure.DavidUtils.Functions.DeserializeDavidFromJsonString(String davidJson)
   
   if I use the newtonsoft to work with the JSON the same error with different file name that missing

**Comment by alenros** [comment]
Can you use other nuget packages in your project?

**Reply by avivs100** [reply]
I am not using any nougat in the project other than the .NET default packages.
If I use the newton soft DLL I get the same error with the missing file System. Streaming.
All other functions of the David class work fine. Like using a constructor, etc.
